keras/src/backend/jax/image.py

```diff
@@ -14,6 +14,20 @@
 )
 
 
+def psnr(image1, image2, max_val):
+    if image1.shape != image2.shape:
+        raise ValueError(
+            "Image shapes must match for PSNR calculation. "
+            f"Received shapes: image1.shape={image1.shape}, "
+            "image2.shape={image2.shape}. "
+        )
+
+    max_val = convert_to_tensor(max_val, image1.dtype)
+    mse = jnp.mean(jnp.square(image1 - image2), axis=(-3, -2, -1))
```

Comment:
The intent of `axis=(-3, -2, -1)` is apparently to exclude the batch axis. But is that necessary? The TF API has no such behavior and returns a scalar.

Reply:
Yup, will remove it!

Reply:
When using axis=(1, 2, 3), the mean squared error (MSE) is computed across all dimensions except the batch dimension. If the user provides batched data, this ensures that PSNR will return a single value instead of one for each batch, which might not be desirable. Instead, it should return a PSNR value for each item in the batch, maintaining consistency with the batch size.

Reply:
TF API has this behavior

[PSNR](https://github.com/tensorflow/tensorflow/blob/5bc9d26649cca274750ad3625bd93422617eed4b/tensorflow/python/ops/image_ops_impl.py#L4211)